tensorflow_probability/python/mcmc/transformed_kernel.py

```diff
@@ -109,6 +105,35 @@ def transformed_log_prob_fn(*state_parts):
     return tlp + sum(ldj_fn(state_parts, event_ndims))
   return transformed_log_prob_fn
 
+def _make_kernel_stack(kernel):
+  kernel_stack = [kernel]
+  while 'target_log_prob_fn' not in kernel.parameters:
+    if 'inner_kernel' not in kernel.parameters:
+      raise ValueError('"None of the nested `inner_kernel`s contains a '
+                       '`target_log_prob_fn`."')
+    kernel = kernel.inner_kernel
+    kernel_stack.append(kernel)
+  return kernel_stack
+
+def get_target_log_prob(kernel):
```

Comment:
This and `update` are (and, I think, should be) public functions.  Add docstrings explaining what they do, arguments, etc.